File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/cior/d/RptOpEmpDO.java
package iih.ci.ord.cior.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.ord.cior.d.desc.RptOpEmpDODesc;
import java.math.BigDecimal;

/**
 * 手术人员 DO数据 
 * 
 */
public class RptOpEmpDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_RPTSUGEMP= "Id_rptsugemp";
	public static final String ID_RPTSUG= "Id_rptsug";
	public static final String ID_EMP= "Id_emp";
	public static final String ID_ROLE= "Id_role";
	public static final String SD_ROLE= "Sd_role";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_rptsugemp() {
		return ((String) getAttrVal("Id_rptsugemp"));
	}	
	public void setId_rptsugemp(String Id_rptsugemp) {
		setAttrVal("Id_rptsugemp", Id_rptsugemp);
	}
	public String getId_rptsug() {
		return ((String) getAttrVal("Id_rptsug"));
	}	
	public void setId_rptsug(String Id_rptsug) {
		setAttrVal("Id_rptsug", Id_rptsug);
	}
	public String getId_emp() {
		return ((String) getAttrVal("Id_emp"));
	}	
	public void setId_emp(String Id_emp) {
		setAttrVal("Id_emp", Id_emp);
	}
	public String getId_role() {
		return ((String) getAttrVal("Id_role"));
	}	
	public void setId_role(String Id_role) {
		setAttrVal("Id_role", Id_role);
	}
	public String getSd_role() {
		return ((String) getAttrVal("Sd_role"));
	}	
	public void setSd_role(String Sd_role) {
		setAttrVal("Sd_role", Sd_role);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_rptsugemp";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_rpt_sug_emp";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(RptOpEmpDODesc.class);
	}
	
}

File: client/iih_stub/iih.ci_stub/iih.ci.ord/cior/d/RptOpEmpDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.ord.cior.d
{
    /// <summary>
    /// RptOpEmpDO 的摘要说明。
    /// </summary>
    public class RptOpEmpDO : BaseDO {

        public RptOpEmpDO() {
        }
		public string Id_rptsugemp {
            get { return getAttrVal<string>("Id_rptsugemp",null); }
            set { setAttrVal<string>("Id_rptsugemp", value); }
        }
		public string Id_rptsug {
            get { return getAttrVal<string>("Id_rptsug",null); }
            set { setAttrVal<string>("Id_rptsug", value); }
        }
		public string Id_emp {
            get { return getAttrVal<string>("Id_emp",null); }
            set { setAttrVal<string>("Id_emp", value); }
        }
		public string Id_role {
            get { return getAttrVal<string>("Id_role",null); }
            set { setAttrVal<string>("Id_role", value); }
        }
		public string Sd_role {
            get { return getAttrVal<string>("Sd_role",null); }
            set { setAttrVal<string>("Sd_role", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_rpt_sug_emp";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_rptsugemp";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.ord.cior.d.RptOpEmpDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_rptsugemp", "Id_rptsug", "Id_emp", "Id_role", "Sd_role"};
        }
        
    }
}
